service/controller/resource/cleanupendpointips/create.go

```diff
@@ -149,28 +154,54 @@ func removeFromEndpointAddressList(addresses []corev1.EndpointAddress, indexesTo
 	return newAddresses
 }
 
+func controlPlanePodForTCNode(node corev1.Node, pods []corev1.Pod) (corev1.Pod, error) {
+	for _, pod := range pods {
+		if pod.Name == node.Name {
+			return pod, nil
```

Comment:
In Go, the convention is usually to make the happy path follow the left side so the last statement should be the return of the matched pod and error cases would go earlier in the function. You'd need to save the pod in a temporary variable and keep track of whether it was found or not with a bool or using a pointer.